### DISCUSSION TITLE
Data writer matches only up to 100 data reader

### DISCUSSION THREAD

**Discussion by MMarcus95** [eProsima/Fast-DDS#4754] [Support]
Hello,

I'm trying to run several data reader and one data writer but I'm getting the following error.
If 100 or less data readers are launched before launching the data writer, the matches between the data readers and the data writer happen as expected.
If instead the data readers are more than 100, the matches do not happen at all. But, If the data writer is launched first, then there is matching.

To test this, I was using the Fastd-DDS version 2.14.0 and the HelloWorld example, by launching one publisher and several subscribers.

Can you explain my why this is happening?

Thanks!

**Comment by JesusPoderoso** [comment]
Hi @MMarcus95, thanks for using Fast DDS.
Please, check the following discussion thread.
- #4178

I am closing this ticket as duplicated. Do not hesitate to reopen it if the suggested solution does not meet your needs.

**Comment by MMarcus95** [comment]
Hi @JesusPoderoso , thanks a lot for the support.

I tried the suggestions proposed in the discussion but I still have the same issue. The only difference is that to set the _readerHistoryMemoryPolicy_ and the _writerHistoryMemoryPolicy_ I'm changing the data reader and data writer qos in this way

```
DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
rqos.endpoint().history_memory_policy = eprosima::fastrtps::rtps::DYNAMIC_RESERVE_MEMORY_MODE;

DataWriterQos wqos = DATAWRITER_QOS_DEFAULT;
wqos.endpoint().history_memory_policy = eprosima::fastrtps::rtps::DYNAMIC_RESERVE_MEMORY_MODE;
```
I also saw that there is the parameter [mutation_tries](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/rtps/attributes/BuiltinAttributes.html#_CPPv4N8eprosima8fastrtps4rtps17BuiltinAttributes14mutation_triesE) but I don't know how to change it from the DDS layer and if it is related to this issue. In the discussion #4178 I think it is not changed.

Do you have any suggestion?

Thanks!

**Comment by JesusPoderoso** [comment]
Hi @MMarcus95, 

>  I don't know how to change it from the DDS layer and if it is related to this issue

the [mutation_tries](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/rtps/attributes/BuiltinAttributes.html#_CPPv4N8eprosima8fastrtps4rtps17BuiltinAttributes14mutation_triesE) can be configured through the participant QoS as follows.

```
DomainParticipantQos participant_qos = PARTICIPANT_QOS_DEFAULT;
participant_qos.wire_protocol().builtin.mutation_tries = 100u; // uint32_t mutation_tries default value is 100
```

**Comment by MMarcus95** [comment]
By changing the mutation_tries to higher value the data writer can now matches more than 100 data reader, thanks!!

Just as a curiosity that might be useful also for others: what actually this parameter does and why it is different from [ResourceLimitedContainerConfig](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/eprosimaExtensions.html#resourcelimitedcontainerconfig)?
